### PR TITLE
Merge client into next

### PR DESCRIPTION
**Pull request by alinhle**
## What does this PR do?
- Merge client into next

### CODE REVIEW

File: patches/relay-compiler/bin/relay-compiler [hunk 1]
```diff
@@ -0,0 +1,10330 @@
+#!/usr/bin/env node
```
Comment (wyattjoh) [inline patches/relay-compiler/bin/relay-compiler]: Is this a compiled file?

File: src/core/server/app/middleware/logging.ts [hunk 2]
```diff
@@ -37,7 +37,7 @@ export const access: RequestHandler = (req, res, next) => {
   next();
 };
 
-export const error: ErrorRequestHandler = (err, req, res, next) => {
+export const error: ErrorRequestHandler = (err, _req, _res, next) => {
```
Comment (wyattjoh) [inline src/core/server/app/middleware/logging.ts]: What's the benefit of this rule for unused params if it just means that to fix bugs with airity, we have to prefix all our params with `_`?

File: tsconfig.json [hunk 3]
```diff
@@ -2,24 +2,40 @@
   "compilerOptions": {
     "target": "es5",
     "module": "commonjs",
-    "esModuleInterop": true,
-    "noImplicitAny": true,
+    "jsx": "react",
```
Comment (wyattjoh) [inline tsconfig.json]: Shouldn't this live in the `src/core/client/tsconfig.json` file?

File: patches/relay-compiler/bin/relay-compiler (hunk 1 above)
Comment (alinhle) [reply patches/relay-compiler/bin/relay-compiler]: Yes

File: tsconfig.json (hunk 3 above)
Comment (alinhle) [reply tsconfig.json]: This was only added for `jest`. The only alternative is to run different instances of `jest` for server and for the client. We can have a discussion around this.

Comment (alinhle) [reply tsconfig.json]: Ah I probably can make it work, without this.

File: src/core/server/app/middleware/logging.ts (hunk 2 above)
Comment (alinhle) [reply src/core/server/app/middleware/logging.ts]: I think we can turn it off, also not sure about its benefits.

File: tsconfig.json (hunk 3 above)
Comment (alinhle) [reply tsconfig.json]: Looks like I don't need this

File: package.json [hunk 4]
```diff
@@ -4,21 +4,39 @@
...
+    "lint:client": "tslint --project ./src/core/client/tsconfig.json",
+    "docz:watch": "docz dev"
   },
   "author": "",
   "license": "Apache-2.0",
   "dependencies": {
+    "@types/relay-runtime": "github:coralproject/patched#types/relay-runtime",
     "apollo-server-express": "^1.3.6",
     "bunyan": "^1.8.12",
+    "chalk": "^2.4.1",
```
Comment (wyattjoh) [inline package.json]: Shouldn't this be in `devDependencies`?

File: package.json [hunk 5]
```diff
@@ -4,21 +4,39 @@
...
+    "lint-fix": "tslint --fix --project ./tsconfig.json && tslint --fix --project ./src/core/client/tsconfig.json",
+    "lint": "npm-run-all --parallel lint:*",
+    "lint:server": "tslint --project ./tsconfig.json",
+    "lint:client": "tslint --project ./src/core/client/tsconfig.json",
+    "docz:watch": "docz dev"
   },
   "author": "",
   "license": "Apache-2.0",
   "dependencies": {
+    "@types/relay-runtime": "github:coralproject/patched#types/relay-runtime",
```
Comment (wyattjoh) [inline package.json]: Shouldn't this be in `devDependencies`?

Comment (alinhle) [reply package.json]: Ah yeah that slipped through!

**Review by wyattjoh — APPROVED** [review]

**Pull request merged**